src/main/java/org/broadinstitute/hellbender/utils/test/SamAssertionUtils.java

```diff
@@ -363,6 +371,44 @@ public static void assertEqualBamFiles(final File resultFile, final File expecte
         }
     }
 
+    // validate that this file has CRAM contents (no matter what the extension)
+    protected static boolean isCRAMFile(final File putativeCRAMFile, final File referenceFile) {
+        boolean isCRAMFile = false;
+        CRAMFileReader cramReader = null;
+        try {
+            // bypass SAMReaderFactory, which will hand back a reader for practically anything
+            cramReader = new CRAMFileReader(
+                    putativeCRAMFile,
+                    new htsjdk.samtools.cram.ref.ReferenceSource(referenceFile)
+            );
+            isCRAMFile = true;
+        }
+        catch (RuntimeException e) {
+            if (!e.getMessage().contains("Unknown file format")) {
+                throw e;
+            }
+            isCRAMFile = false;
+        }
+        finally {
+            if (null != cramReader) {
+                cramReader.close();
```

Comment:
Its not closeable even though it has a close method (closeableness is conferred on it via PrimitiveSamReaderToSamReaderAdapter only when its retrieved through SAMReaderFactory).